.github/workflows/lint-test.yml

```diff
@@ -33,144 +38,228 @@ jobs:
     # https://pdm-project.org/en/latest/usage/lockfile/
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v4
-
-    # Instead of setup-env, we call the cache-pdm action here directly.
-    # This avoids having to install PDM, only to find out the cache is already up to date sometimes.
-    - name: Configure cache
-      uses: ./.github/actions/cache-pdm
-      with:
-        env-already-initialized: false
-
+      - uses: actions/checkout@v4
+
+      - name: Set up uv to lock dependencies
+        id: setup-python
+        uses: astral-sh/setup-uv@v6
+        with:
+          version: ${{ env.USE_UV_VERSION }}
+          enable-cache: true
+
+      - name: Restore uv lock
+        uses: actions/cache/restore@v4
+        id: cache-uv-lock
+        with:
+          key: uv-lock-resolved-${{ env.UV_RESOLUTION }}
+          restore-keys: |
+            uv-lock-resolved-${{ env.UV_RESOLUTION }}
+          path: uv.lock
+          enableCrossOsArchive: true
+
+      - name: Check if uv.lock changed
+        id: uv-lock-changed
+        continue-on-error: true
+        if: steps.cache-uv-lock.outputs.cache-hit == 'true' || steps.cache-uv-lock.outputs.cache-hit == 'false'
+        run: |
+          uv lock --check
+
+      - name: Lock dependencies
+        if: steps.uv-lock-changed.outcome == 'failure' || steps.cache-uv-lock.outputs.cache-hit != 'true' && steps.cache-uv-lock.outputs.cache-hit != 'false'
+        id: lock-uv-deps
+        env:
+          UV_LOCKED: 0
+        run: uv lock --refresh
+
+      - uses: actions/cache/save@v4
+        if: steps.lock-uv-deps.outcome == 'success'  && steps.uv-lock-changed.outcome == 'failure'
+        name: upload-uv-lock
+        with:
+          key: uv-lock-resolved-${{ env.UV_RESOLUTION}}-${{hashFiles('uv.lock')}}
```

Comment:
regarding `hashFiles` in general: assuming GitHub still hasn't fixed this, `hashFiles` can differ on windows vs macos/linux due to line endings, which is why `.gitattributes` contains pyproject.toml. That one is now definitely unnecessary, but I'm not sure if we'd need to add `uv.lock` to it.